classes/order/Order.php

```diff
@@ -1173,7 +1173,7 @@ public static function getByCartId($id_cart)
     {
         $id_order = (int) self::getIdByCartId((int) $id_cart);
 
-        return ($id_order > 0) ? new self($id_order) : null;
+        return ($id_order > 0) ? new Order($id_order) : null;
```

Comment:
```suggestion
        return ($id_order > 0) ? new static($id_order) : null;
```

If you want to return the same instance type, use `static` instead

Reply:
For example: OrderSlip::create() requires an instance of "Order". And many other places.

Reply:
new static($id_order) well